### ISSUE TITLE
Manually entering date is hard/not possible

### ISSUE THREAD

**Issue by jvolker** [strapi/strapi#10510]
### Describe the bug

In a date picker, I'm trying to type in the year of the date string manually. When trying to delete the year (in order to change it to a different year) from eg. `06/03/2021` the year string fills up automatically with preceding 0s to 4 digits after a short moment:`06/03/0202`

I would like to enter a date from the 1950s. Using the UI instead of typing, to get back 70 years month by month would take a long time.

### Steps to reproduce the behavior

1. Open a collection type that contains a date picker.
2. Click on a day to prefill the date string
3. Try to change the year of the date string (by deleting the year from the end of the string) 

### Expected behavior

The date string should be editable without being overwritten automatically. Only after clicking outside the input field or pressing enter, a correction of the string should apply.

### Screenshots

![image](https://user-images.githubusercontent.com/546852/122535125-893f6c80-d023-11eb-9bf8-e88cfbde0ec6.png)

### System

- Node.js version: v12.20.1
- NPM version: 
- Strapi version: v3.6.3
- Database: Postgres
- Operating system: Docker/Ubuntu

### Additional context

https://github.com/strapi/buffet/issues/68

**Comment by crussell42** [comment]
Still an issue with strapi 3.6.5 in Sept 2021.

No matter how good the strapi concept is, this (seemingly minor inconvenience) for
a react developer who does not care to test using dates more than 2 months in the past;
is a flat out (I cant use this damn thing) for users trying to actually 
publish and use strapi as a CMS. 

I really want to see strapi succeed so I think the severity:low / feature request is wrong.
This is not a feature request to "users".

**Comment by neahan** [comment]
agreed. this really should be a higher priority.

**Comment by oscarsandberg** [comment]
Is this issue still open? If it is, can I contribute to it?

**Comment by cursomeancore** [comment]
I think this is very necessary. Is there a way I can do a quick fix while it is being fixed? Unless the value can be written. Thanks

**Comment by oscarsandberg** [comment]
I haven't looked into this that much, so it is difficult to say if there is a quick fix. I'll start to look at this today.

**Comment by MattieBelt** [comment]
Hey @oscarsandberg,

Thanks for offering to contribute, but keep in mind that the repo is currently under a [community pr change freeze](https://github.com/strapi/strapi/issues/10695) and the team is implementing a new Design System for `v4`.
So I'm not sure if it's worth it to focus on a PR for the `v3` or Buffet.js component lib, as the `v4` beta has been released yesterday and it seems that [the new DateTimePicker](https://design-system-git-develop-strapijs.vercel.app/?path=/story/design-system-components-datepicker--base) still not allows for manual dates.

**Comment by oscarsandberg** [comment]
Since the current date picker will be deprecated in a relatively near future, I won't fix this. 

@cursomeancore You can use the `wait` prop to increase the time till the date picker autofills a date.

**Comment by jobe451** [comment]
Is there an issue on the current time pick I can track? I would like to upgrade to a newer version as soon as that is out. Especially the current time picker is maximized for inconvenience.

**Comment by Kaherdin** [comment]
It's still not working fine with v4 ! I cannot type the date neither the hour with my keyboard. DatePicker UX = 1/10 
Please, improve this :)

**Comment by gu-stav** [comment]
Related to https://github.com/strapi/strapi/issues/12175 .

**Comment by Shrimpstronaut** [comment]
Hey everyone, I've been following the progress of the date- / time-picker for a while. I'm unable to track the progress of this improvement and wanted to check if this is still actively being worked on. 
I would happily assist with providing a PR, as this is a big pain in the current version of Strapi.

Thanks

**Comment by gu-stav** [comment]
@Shrimpstronaut Thanks for your message. For now, we have stopped working on it, since it is not an easy fix with the current implementation in the design system. I do understand your frustration about it and I'm happy to assist you, in case you would like to work on it (either the date or the time picker).
